grade-management-new/GradeManagement.Client/Components/ExerciseDataGrid.razor

```diff
@@ -0,0 +1,64 @@
+﻿@using GradeManagement.Client.Pages
+@inject HttpClient Http
+@inject CrudSnackbarService SnackbarService
+
+<MudDataGrid T="Exercise" Items="Values" EditMode="DataGridEditMode.Form" ReadOnly="false" CommittedItemChanges="@CommittedItemChanges" Style="width: 100%">
+    <Columns>
+        <PropertyColumn Property="x => x.Id" Title="Id" IsEditable="false"/>
+        <PropertyColumn Property="x => x.Name" Title="Name"/>
+        <PropertyColumn Property="x => x.GithubPrefix" Title="Github Prefix"/>
+        <TemplateColumn Title="Actions">
+            <CellTemplate>
+                <MudStack Row>
+                    <MudIconButton Icon="@Icons.Material.Filled.Edit" Color="Color.Primary" Size="Size.Small" OnClick="@(() => StartEditing(context))"/>
+                    <MudIconButton Icon="@Icons.Material.Filled.Delete" Color="Color.Error" Size="Size.Small" OnClick="@(() => DeleteItem(context.Item))"/>
+                </MudStack>
+            </CellTemplate>
+        </TemplateColumn>
+    </Columns>
+</MudDataGrid>
+
+
+@code {
+    [Parameter] public List<Exercise> Values { get; set; }
+
+    string BaseUrl = Endpoints.EXERCISES;
+
+    private async Task StartEditing(CellContext<Exercise?> context)
+    {
+        await context.Actions.StartEditingItemAsync();
+        StateHasChanged();
+    }
+
+    private async Task CommittedItemChanges(Exercise items)
+    {
+        var response = await Http.PutAsJsonAsync($"{BaseUrl}/{items.Id}", items);
+        if (!response.IsSuccessStatusCode)
```

Comment:
A hibakezelés is abba a service osztályba illik, ahol hibadobást szoktak végezni majd ezt lekezelni egy központi helyen, hogy kelljen minden komponensben ezt a logikát duplikálni